packages/crypto/src/crypto.ts

```diff
@@ -0,0 +1,389 @@
+): Uint8Array => {
+  const combinedLength =
+    HPKE_VERSION.length + suiteId.length + label.length + ikm.length;
+  const ret = new Uint8Array(combinedLength);
+  let offset = 0;
+
+  ret.set(HPKE_VERSION, offset);
+  offset += HPKE_VERSION.length;
+
+  ret.set(suiteId, offset);
+  offset += suiteId.length;
+
+  ret.set(label, offset);
+  offset += label.length;
+
+  ret.set(ikm, offset);
+
+  return ret;
+};
+
+/**
+ * Build labeled info for HKDF operations.
+ *
+ * @param {Uint8Array} label - The label to use.
+ * @param {Uint8Array} info - Additional information.
+ * @param {Uint8Array} suite_id - The suite identifier.
+ * @param {number} len - The output length.
+ * @returns {Uint8Array} - The labeled info.
+ */
+const buildLabeledInfo = (
+  label: Uint8Array,
+  info: Uint8Array,
+  suite_id: Uint8Array,
+  len: number
+): Uint8Array => {
+  const ret = new Uint8Array(
+    9 + suite_id.byteLength + label.byteLength + info.byteLength
+  );
+  ret.set(new Uint8Array([0, len]), 0); // this isn’t an error, we’re starting at index 2 because the first two bytes should be 0. See <https://github.com/dajiaji/hpke-js/blob/1e7fb1372fbcdb6d06bf2f4fa27ff676329d633e/src/kdfs/hkdf.ts#L41> for reference.
+  ret.set(HPKE_VERSION, 2);
+  ret.set(suite_id, 9);
+  ret.set(label, 9 + suite_id.byteLength);
+  ret.set(info, 9 + suite_id.byteLength + label.byteLength);
+  return ret;
+};
+
+/**
+ * Convert a BigInt to a hexadecimal string of a specific length.
+ */
+const bigIntToHex = (num: bigint, length: number): string => {
+  const hexString = num.toString(16);
+  if (hexString.length > length) {
+    throw new Error(
+      `number cannot fit in a hex string of ${length} characters`
+    );
+  }
+  return hexString.padStart(length, "0");
+};
+
+/**
+ * Uncompress a raw public key.
+ */
+
+const uncompressRawPublicKey = (rawPublicKey: Uint8Array): Uint8Array => {
+  // point[0] must be 2 (false) or 3 (true).
+  // this maps to the initial "02" or "03" prefix
+  const lsb = rawPublicKey[0] === 3;
+  const x = BigInt("0x" + uint8ArrayToHexString(rawPublicKey.subarray(1)));
+
+  // https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.186-4.pdf (Appendix D).
+  const p = BigInt(
+    "115792089210356248762697446949407573530086143415290314195533631308867097853951"
+  );
+  const b = BigInt(
+    "0x5ac635d8aa3a93e7b3ebbd55769886bc651d06b0cc53b0f63bce3c3e27d2604b "
+  );
+  const a = p - BigInt(3);
+
+  // Now compute y based on x
+  const rhs = ((x * x + a) * x + b) % p;
+  let y = modSqrt(rhs, p);
+  if (lsb !== testBit(y, 0)) {
+    y = (p - y) % p;
+  }
+
+  if (x < BigInt(0) || x >= p) {
+    throw new Error("x is out of range");
+  }
+
+  if (y < BigInt(0) || y >= p) {
+    throw new Error("y is out of range");
+  }
+
+  var uncompressedHexString = "04" + bigIntToHex(x, 64) + bigIntToHex(y, 64);
+  return uint8ArrayFromHexString(uncompressedHexString);
+};
+/**
+ * Compute the modular square root using the Tonelli-Shanks algorithm.
+ */
+
```

Comment:
tiny nit: let's add a newline after line 257, and remove this extra newline here